src/rgw/rgw_sync_module_aws.cc

```diff
@@ -0,0 +1,1728 @@
+#include "common/errno.h"
+
+#include "rgw_common.h"
+#include "rgw_coroutine.h"
+#include "rgw_sync_module.h"
+#include "rgw_data_sync.h"
+#include "rgw_sync_module_aws.h"
+#include "rgw_cr_rados.h"
+#include "rgw_rest_conn.h"
+#include "rgw_cr_rest.h"
+#include "rgw_acl.h"
+
+#include <boost/asio/yield.hpp>
+
+#define dout_subsys ceph_subsys_rgw
+
+
+#define DEFAULT_MULTIPART_SYNC_PART_SIZE (32 * 1024 * 1024)
+
+static string default_target_path = "rgw-${zonegroup}-${sid}/${bucket}";
+
+static string get_key_oid(const rgw_obj_key& key)
+{
+  string oid = key.name;
+  if (!key.instance.empty() &&
+      !key.have_null_instance()) {
+    oid += string(":") + key.instance;
+  }
+  return oid;
+}
+
+static string obj_to_aws_path(const rgw_obj& obj)
+{
+  string path = obj.bucket.name + "/" + get_key_oid(obj.key);
+
+
+  return path;
+}
+
+/*
+
+   json configuration definition:
```

Comment:
@yehudasa this json configuration is fancy and powerful.  could you give me a simple example how to create a zone. I have tried following command, but can't get a valid tier config. 

```
./bin/radosgw-admin  zone create --rgw-zonegroup=test-sync-cos-group --rgw-zone=zone1 --endpoints=127.0.0.1:8005 --access-key=accesskey  --secret=secretkey --tier-type=aws  --tier-config="connection=\"{\"access_key\": \"accesskey\", \"secret\": \"secretkey\", \"endpoint\":\"127.0.0.1:8006\", \"host_style\": \"path\"},target_path=\"rgw-${zonegroup}-${sid}/${bucket}\""  --debug_rgw=20
```

"zone get" shows : 
```
    "tier_config": {
        " host_style: path}": "",
        "target_path": "rgw--/"
    },
```

thank you.

Reply:
@yehudasa hi, I have fix this bug, see https://github.com/yehudasa/ceph/pull/9

btw, I got a core-dump when I run "data sync run" command. my colleague sends a PR to fix it. https://github.com/ceph/ceph/pull/20611

I've only done some simple tests, and now large and small files can be successfully synchronized.


![image](https://user-images.githubusercontent.com/1787766/36719323-b3ce764a-1bdf-11e8-96a0-fa3e0d1e59ef.png)